diff --git a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -89,7 +89,7 @@
 [dev-dependencies.tempfile]
 version = "=3.27.0"
 
-[target.'cfg(any(target_os = "linux", target_os = "macos"))'.dependencies.libc]
+[target.'cfg(any(target_os = "linux", target_os = "macos", target_os = "emscripten"))'.dependencies.libc]
 version = "=0.2.186"
 
 [target."cfg(windows)".dependencies.windows-sys]
