diff --git a/src/lib.rs b/src/lib.rs
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -25,21 +25,21 @@
 #[cfg(any(target_os = "macos", target_os = "ios"))]
 use mac as sys;
 
-#[cfg(target_arch = "wasm32")]
+#[cfg(all(target_arch = "wasm32", not(target_os = "emscripten")))]
 mod wasm;
-#[cfg(target_arch = "wasm32")]
+#[cfg(all(target_arch = "wasm32", not(target_os = "emscripten")))]
 use wasm as sys;
 
 #[cfg(not(any(
     target_os = "windows",
     target_os = "macos", target_os = "ios",
-    target_arch = "wasm32"
+    all(target_arch = "wasm32", not(target_os = "emscripten"))
 )))]
 mod lin;
 #[cfg(not(any(
     target_os = "windows",
     target_os = "macos", target_os = "ios",
-    target_arch = "wasm32"
+    all(target_arch = "wasm32", not(target_os = "emscripten"))
 )))]
 use lin as sys;
 
